### PR TITLE
RSDK-3724 - Exclude credential authorization header from ViamChannel::WebRTC based connections

### PR DESCRIPTION
**Pull request by edaniels**
Tested manually by making sure direct pre authorized connections still work and pass through headers while webrtc ones do not (via app or mdns found)

### CODE REVIEW

File: src/rpc/dial.rs [hunk 1]
```diff
@@ -143,13 +144,15 @@ impl Service<http::Request<BoxBody>> for ViamChannel {
     fn poll_ready(&mut self, cx: &mut TaskContext<'_>) -> Poll<Result<(), Self::Error>> {
         match self {
             Self::Direct(channel) => channel.poll_ready(cx),
+            Self::DirectPreAuhtorized(channel) => channel.poll_ready(cx),
```
Comment (edaniels) [inline src/rpc/dial.rs:147]: I don't love this but it seemed the be the simplest approach in order to get the Result type to work better without adding an Either or having it be a `dyn tonics specific service type with associations that is really annoying to type out`

File: src/rpc/dial.rs [hunk 2]
```diff
@@ -143,13 +144,15 @@ impl Service<http::Request<BoxBody>> for ViamChannel {
...
             Self::WebRTC(_channel) => Poll::Ready(Ok(())),
         }
     }
 
     fn call(&mut self, request: http::Request<BoxBody>) -> Self::Future {
         match self {
             Self::Direct(channel) => Box::pin(channel.call(request)),
+            Self::DirectPreAuhtorized(channel) => Box::pin(channel.call(request)),
```
Comment (cheukt) [inline src/rpc/dial.rs:155]: PreAuhtorized -> PreAuthorized

**Review by stuqdog — APPROVED** [review]
LGTM!

File: src/rpc/dial.rs (hunk 2 above)
Comment (edaniels) [reply src/rpc/dial.rs:155]: whoops

**Pull request merged**